### PR TITLE
Compile and warning fixes for Visual Studio 15.5.4 (C++17)

### PR DESCRIPTION
**Pull request by igfraga**

### CODE REVIEW

File: libfive/src/solve/bounds.cpp [hunk 1]
```diff
@@ -42,8 +42,9 @@ Region<3> findBounds(IntervalEvaluator* eval)
```
Comment (igfraga) [inline libfive/src/solve/bounds.cpp]: This one was causing the compiler to crash. Breaking it up like this fixed it, but of course there are other ways to do this.

**Comment by igfraga** [comment]
These changes fix all warnings except for:
/wd4267") # int64 -> int
/wd4244") # float -> int
/wd4305") # double -> float

We could fix those too if you like, but they are minor and require a lot of little changes.

File: libfive/src/solve/bounds.cpp (hunk 1 above)
Comment (speps) [reply libfive/src/solve/bounds.cpp]: I was about to look at this one, how did you find where it crashed the compiler?

Comment (igfraga) [reply libfive/src/solve/bounds.cpp]: let's say it was a binary-search commenting out of code

**Comment by mkeeter** [comment]
Thanks!

I checked out this branch and added a few style tweaks, which you can see in the branch [`libfive/nTopology-shared/ntop`](https://github.com/libfive/libfive/tree/nTopology-shared/ntop).

Can you confirm that this still works before I merge it in?
(let's keep debugging the Eigen assertion via email)

**Comment by speps** [comment]
I also have a CL to make it easy to get the dependencies, working on updating documentation to use `vcpkg` now. That means we could add support for VS2017 and it would be as easy as the other platforms. Hopefully we can find a solution to have a version of Studio on Windows too.

**Comment by mkeeter** [comment]
@speps Studio will probably have to use MinGW (`gcc`), rather than MSVC, because of the Guile dependency.  Even then, I've been trying to build a native Guile in MinGW, with [mixed results](https://lists.gnu.org/archive/html/guile-devel/2018-01/msg00008.html).

Still, we should definitely get the core `libfive` working under MSVC, which is why I appreciate your and @igfraga 's input 😃

**Comment by igfraga** [comment]
Tested your branch, looks good. Feel free to close this PR or update it.

**Pull request merged**